src/main/java/thrift/logic/commands/CloneCommand.java

```diff
@@ -92,30 +103,55 @@ public CommandResult execute(Model model, TransactionListPanel transactionListPa
         }
 
         Transaction transactionToClone = lastShownList.get(targetIndex.getZeroBased());
+        Date originalDate = transactionToClone.getDate().getDate();
+
         frequencyCalendarField = occurrence.getFrequencyCalendarField();
+        assert (frequencyCalendarField == Calendar.DATE) || (frequencyCalendarField == Calendar.WEEK_OF_YEAR)
+                || (frequencyCalendarField == Calendar.MONTH) || (frequencyCalendarField == Calendar.YEAR)
+                : "Frequency of occurrence for clones not converted to a valid Calendar field";
+
+        for (int i = 0; i <= occurrence.getNumOccurrences(); i++) {
+            // If cloning multiple times, skip step to clone a transaction on date of original transaction
+            if (occurrence.getNumOccurrences() > 0 && i == 0) {
+                continue;
+            }
 
-        for (int i = 0; i < occurrence.getNumOccurrences(); i++) {
+            // Produce the date attribute of cloned transaction
             Calendar calendar = Calendar.getInstance();
+            if (occurrence.getNumOccurrences() > 0) {
+                calendar.setTime(originalDate);
+            }
             calendar.add(frequencyCalendarField, i);
             String date = DATE_FORMATTER.format(calendar.getTime());
 
             Transaction clonedTransaction = createClonedTransaction(transactionToClone, date);
             clonedTransactionList.add(clonedTransaction);
             if (clonedTransaction instanceof Expense) {
                 model.addExpense((Expense) clonedTransaction);
-            } else if (clonedTransaction instanceof Income) {
+            } else {
+                assert clonedTransaction instanceof Income : "Transaction to Clone not of type Expense or Income";
                 model.addIncome((Income) clonedTransaction);
             }
+
             // Use null comparison instead of requireNonNull(transactionListPanel) as current JUnit tests are unable to
             // handle JavaFX initialization
             if (transactionListPanel != null && model.isInView(clonedTransaction)) {
                 transactionListPanel.getTransactionListView().scrollTo(model.getFilteredTransactionList().size() - 1);
             }
+
+            // For single Clone operation, end Cloning after one iteration
+            if (occurrence.getNumOccurrences() == 0) {
+                logger.info("CLONED transaction [ " + clonedTransaction + " ] [1 TIME]");
+                break;
+            }
+
+            logger.info("CLONED transaction [ " + clonedTransaction + " ] [" + i + " TIMES]");
         }
 
+        int actualTimesCloned = (occurrence.getNumOccurrences() == 0) ? 1 : occurrence.getNumOccurrences();
+        String actualFrequency = (occurrence.getNumOccurrences() == 0) ? "for today" : occurrence.getFrequency();
```

Comment:
Instead of using 2 ternary operators, maybe just dilute it to:
```
int actualTimesCloned = occurrence.getNumOccurrences();
String actualFrequency = occurrence.getFrequency();
if (occurrence.getNumOccurrences() == 0) {
    actualTimesCloned = 1;
    actualFrequency = "for today";
}
```